skfda/exploratory/visualization/_baseplot.py

```diff
@@ -0,0 +1,82 @@
+"""BasePlot Module.
+
+This module contains the abstract class of which inherit all
+the visualization modules, containing the basic functionality
+common to all of them.
+"""
+
+from abc import ABC, abstractmethod
+from typing import List, Optional, Sequence, Union
+
+import matplotlib.pyplot as plt
+from matplotlib.artist import Artist
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+
+from ._utils import _figure_to_svg
+
+
+class BasePlot(ABC):
+    """
+    BasePlot class.
+
+    Attributes:
+        artists: List of Artist objects corresponding
+            to every instance of our plot. They will be used to modify
+            the visualization with interactivity and widgets.
+        fig: Figure over with the graphs are plotted.
+        axes: Sequence of axes where the graphs are plotted.
+    """
+
+    @abstractmethod
+    def __init__(
+        self,
+    ) -> None:
+        self.artists: List[Artist] = []
+        self.fig: Optional[Figure] = None
+        self.axes: Sequence[Axes] = []
+
+    @abstractmethod
+    def plot(
+        self,
+    ) -> Figure:
+        """
+        Abstract method used to plot the object and its data.
+
+        Returns:
+            Figure: figure object in which the displays and
+                widgets will be plotted.
+        """
+        pass
+
+    @abstractmethod
+    def n_samples(self) -> int:
+        """Get the number of instances that will be used for interactivity."""
+        pass
+
+    @abstractmethod
+    def set_figure_and_axes(
```

Comment:
So, which differences should be in subclasses? Is this the right abstraction?

Reply:
it was a way to call all the initialization axes and fig methods from utils the same way, some times we initialize it with some parameters other times with others, that is the reason for this method

Reply:
Can you be more concrete?

Reply:
for example the boxplot uses _get_figure_and_axes, _set_figure_layout_for_fdata to initialize the axes, while FPCA only uses _get_figure_and_axes and does a condition. In this way i encapsulate in every file the functionality to initialize axes for a clearer understanding

Reply:
Where is this called? In plot?

Reply:
we always initialize the axes in init, you can see in the outliergram interactive branch

Reply:
Then why do we need this method? Couldn't this be done in init (which is different for each plot type) and avoid exposing another public method?

Reply:
ok, so we do it in init, it is unnecessary